packages/core/src/App/Containers/PasswordSelectionModal/password-selection-modal.jsx

```diff
@@ -6,103 +6,128 @@ import { getErrorMessages, redirectToSignUp } from '@deriv/shared';
-        <Text as='p' weight='bold' className='account-signup__heading' align='center'>
-            <Localize i18n_default_text='Keep your account secure with a password' />
-        </Text>
-        <Field name='password'>
-            {({ field }) => (
-                <PasswordMeter
-                    input={pw_input}
-                    has_error={!!(touched.password && errors.password)}
-                    custom_feedback_messages={getErrorMessages().password_warnings}
-                >
-                    <PasswordInput
-                        {...field}
-                        autoComplete='new-password'
-                        className='account-signup__password-field'
-                        label={localize('Create a password')}
-                        error={touched.password && errors.password}
-                        required
-                        value={values.password}
-                        onBlur={handleBlur}
-                        onChange={e => {
-                            const input = e.target;
-                            setFieldTouched('password', true);
-                            if (input) updatePassword(input.value);
-                            handleChange(e);
-                        }}
-                        input_id='dt_core_account-signup-modal_account-signup-password-field'
-                    />
-                </PasswordMeter>
-            )}
-        </Field>
+const PasswordSelectionModal = observer(
+    ({
+        api_error,
+        errors,
+        handleBlur,
+        handleChange,
+        is_appstore,
+        isModalVisible,
+        isSubmitting,
+        touched,
+        pw_input,
+        setFieldTouched,
+        updatePassword,
+        values,
+    }) => {
+        const { ui } = useStore();
 
-        <Text as='p' size='xxs' className='account-signup__subtext' align='center'>
-            <Localize i18n_default_text='Strong passwords contain at least 8 characters, combine uppercase and lowercase letters, numbers, and symbols.' />
-        </Text>
-        <SignupSeparatorContainer />
-        {api_error ? (
-            <React.Fragment>
-                <Text
-                    as='p'
-                    size='xxs'
-                    color='loss-danger'
-                    className='account-signup__subtext account-signup__subtext--error'
-                    align='center'
-                >
-                    {api_error}
+        const { is_mobile } = ui;
+
+        React.useEffect(() => {
+            RudderStack.track(
+                'ce_virtual_signup_form',
+                {
+                    action: 'password_screen_opened',
+                    form_name: is_mobile ? 'virtual_signup_web_mobile_default' : 'virtual_signup_web_desktop_default',
+                },
+                {
+                    is_anonymous: true,
+                }
+            );
+
+            //eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Instead of using this disable rule can we add the necessary dependencies to the useEffect?